### ISSUE TITLE
Datepicker configuration not working

### ISSUE THREAD

**Issue by AlexPeret** [dotnet-websharper/jqueryui#9]
hi!

I've tried to change a few options on Datepicker component by using the configuration type, but with no effect.

For instance, none of the options seems to work:
```
...
let config = DatepickerConfiguration()
config.ShowOn <- "focus"
config.DateFormat <- "dd/mm/yy"
config.ChangeMonth <- true
config.ChangeYear <- true
config.AutoSize <- true

let el = Div []
let picker = Datepicker.New(el,config)
...
```

Thanks,
Alex

**Comment by AlexPeret** [comment]
please, ignore this issue. It is working when running on a separate project.

**Comment by Tarmil** [comment]
Have you found why it didn't work in your original project? It sounds like there might still be something fishy going on.

---

As a side note, you can set properties as if they were named arguments to the constructor, which I think looks much nicer:

```fsharp
let config =
    DatePickerConfiguration(
        ShowOn = "focus",
        DateFormat = "dd/mm/yy",
        ChangeMonth = true,
        ChangeYear = true,
        AutoSize = true
    )
```

**Comment by AlexPeret** [comment]
yes, I was too fast on closing it.

Turns out the OnSelect delegate is breaking. Below, the snippet I'm using to test:

```
    open System
    open WebSharper.Html.Client
    open WebSharper.JQuery
    open WebSharper.JQueryUI
    let PickDateTest () =

        let config = DatepickerConfiguration()
        //config.ShowOn <- "focus"
        config.DateFormat <- "dd/mm/yy"
        config.ChangeMonth <- true
        config.ChangeYear <- true
        config.AutoSize <- true

        //let elDiv = div[]
        //let picker = Datepicker.New(elDiv.Dom,config)
        let el = Div []
        let picker = Datepicker.New(el,config)

        // this breaks the component
        picker.OnSelect (fun dt elem ->
            Console.Log "OnSelect"
        )

        let pickerDoc = 
            el.Dom
            |> View.Const
            |> View.Map (fun elem ->
                picker.Render ()
                Doc.Static elem)
            |> Doc.EmbedView 

        pickerDoc

```
Thanks for the tip, looks better indeed.
Alex

**Comment by AlexPeret** [comment]
I'm not sure about it, but maybe the line below from Datepicker.fs is replacing the original configuration structure by a new one:

```
[<Inline "jQuery($this.element.Dom).datepicker({onSelect: function (x,y) {($f(x))(y);}})">]
member private this.onSelect(f : string -> Datepicker -> unit) = ()
```
this: **{onSelect: function (x,y) {($f(x))(y);}}**

**Comment by AlexPeret** [comment]
hi!

I've create a fork of this project and made some changes to fix this issue.

You can find them at: 
https://github.com/AlexPeret/jqueryui/commit/f9d5187eda391de28cebe1d4c66af2dfc5dc6e24

Let me know if you need any clarification.

Regards,
Alex

**Comment by Tarmil** [comment]
Hi @AlexPeret, feel free to post a pull request here from your fork! I compiled it locally and looked at the test project, it seems good.

**Comment by AlexPeret** [comment]
Hi @Tarmil! it's done.

I'm not familiar with Github and couldn't find a way to remove unnecessary files from the Pull Request.

Please, let me know if you need me to make any change.

**Comment by Tarmil** [comment]
Fixed in #12.
